packages/test/test-version-utils/src/describeCompat.ts

```diff
@@ -0,0 +1,84 @@
+import { configList } from "./compatConfig";
+import { CompatKind, baseVersion, driver, r11sEndpointName, tenantIndex } from "./compatOptions";
+import { getVersionedTestObjectProvider } from "./compatUtils";
+
+/*
+ * Mocha Utils for test to generate the compat variants.
+ */
+function createCompatSuite(
+    tests: (this: Mocha.Suite, provider: () => ITestObjectProvider) => void,
+    compatFilter?: CompatKind[],
+) {
+    return function (this: Mocha.Suite){
+        let configs = configList.value;
+        if (compatFilter !== undefined) {
+            configs = configs.filter((value) => compatFilter.includes(value.kind));
+        }
+
+        for (const config of configs) {
+            describe(config.name, function () {
+                let provider: TestObjectProvider;
+                let resetAfterEach: boolean;
+                before(async function () {
+                    provider = await getVersionedTestObjectProvider(
+                        baseVersion,
+                        config.loader,
+                        {
+                            type: driver,
+                            version: config.driver,
+                            config: {
+                                r11s: { r11sEndpointName },
+                                odsp: { tenantIndex },
+                            },
+                        },
+                        config.containerRuntime,
+                        config.dataRuntime,
+                    );
+                    Object.defineProperty(this,"__fluidTestProvider",{get: ()=>provider});
+                });
+                tests.bind(this)((reset: boolean = true) => {
+                    if (reset) {
+                        resetAfterEach = true;
+                    }
+                    return provider;
+                });
+                afterEach(function (done:Mocha.Done) {
+                    done(getUnexpectedLogErrorException(provider.logger, "Use itExpects to specify expected errors. "));
+                    if (resetAfterEach) {
+                        provider.reset();
+                    }
+                });
+            });
+        }
+    };
+}
+
+export type DescribeCompatSuite=
+    (name: string,
+    tests: (
+        this: Mocha.Suite,
+        provider: (resetAfterEach?: boolean) => ITestObjectProvider) => void
+    ) => Mocha.Suite | void
+
+export type DescribeCompat = DescribeCompatSuite & Record<"skip" | "only" | "noCompat", DescribeCompatSuite>;
+
+function createCompatDescribe(compatFilter?: CompatKind[]): DescribeCompat{
+    const d: DescribeCompat =
+        (name, tests) => describe(name, createCompatSuite(tests, compatFilter));
+    d.skip = (name, tests) => describe.skip(name, createCompatSuite(tests, compatFilter));
+    d.only = (name, tests) => describe.only(name, createCompatSuite(tests, compatFilter));
+    d.noCompat = (name, tests) => describe(name, createCompatSuite(tests, [CompatKind.None]));
```

Comment:
This property will allow running any tests with noCompat, rather than having to change the describe to describeNoCompat and fix up the imports.